[flake]: TestBatchedAssetIssuance/minting_with_cancellation flake

This test was observed failing flakily in a CI run for #1881 under the "unit tests (unit-cover)" job:

```
--- FAIL: TestBatchedAssetIssuance (87.49s)
    --- FAIL: TestBatchedAssetIssuance/minting_with_cancellation (62.30s)
        test_sqlite.go:11: Creating new SQLite DB handle for testing: /tmp/TestBatchedAssetIssuanceminting_with_cancellation1519908230/001/tmp.db
        planter_test.go:382: Missing expected seedling: (*tapgarden.Seedling)(0xc0008c6840)({
              AssetVersion: (asset.Version) 0,
              AssetType: (asset.Type) 1,
              AssetName: (string) (len=64) "65fff78845e603d0788ee37a622e6f28248c433fbd70691dafb6dcb8754c7b76",
              Meta: (*proof.MetaReveal)(0xc0008e1ce0)({
                Type: (proof.MetaType) 0,
                Data: ([]uint8) (len=32 cap=32) {
                  00000000  65 ff f7 88 45 e6 03 d0  78 8e e3 7a 62 2e 6f 28  |e...E...x..zb.o(|
                  00000010  24 8c 43 3f bd 70 69 1d  af b6 dc b8 75 4c 7b 76  |$.C?.pi.....uL{v|
                },
                DecimalDisplay: (fn.Option[uint32]) {
                  isSome: (bool) false,
                  some: (uint32) 0
                },
                UniverseCommitments: (bool) false,
                CanonicalUniverses: (fn.Option[[]net/url.URL]) {
                  isSome: (bool) false,
                  some: ([]url.URL) <nil>
                },
                DelegationKey: (fn.Option[github.com/decred/dcrd/dcrec/secp256k1/v4.PublicKey]) {
                  isSome: (bool) false,
                  some: (secp256k1.PublicKey) {
                    x: (secp256k1.FieldVal) {
                      n: ([10]uint32) (len=10 cap=10) {
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0
                      }
                    },
                    y: (secp256k1.FieldVal) {
                      n: ([10]uint32) (len=10 cap=10) {
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0,
                        (uint32) 0
                      }
                    }
                  }
                },
                UnknownOddTypes: (tlv.TypeMap) <nil>
              }),
              Amount: (uint64) 1,
              GroupInfo: (*asset.AssetGroup)(<nil>),
              EnableEmission: (bool) false,
              SupplyCommitments: (bool) false,
              DelegationKey: (fn.Option[github.com/lightningnetwork/lnd/keychain.KeyDescriptor]) {
                isSome: (bool) false,
                some: (keychain.KeyDescriptor) {
                  KeyLocator: (keychain.KeyLocator) {
                    Family: (keychain.KeyFamily) 0,
                    Index: (uint32) 0
                  },
                  PubKey: (*secp256k1.PublicKey)(<nil>)
                }
              },
              GroupAnchor: (*string)(<nil>),
              updates: (tapgarden.SeedlingUpdates) (cap=1) 0xc0000c7570,
              ScriptKey: (asset.ScriptKey) {
                PubKey: (*secp256k1.PublicKey)(0xc0008aae60)({
                  x: (secp256k1.FieldVal) {
                    n: ([10]uint32) (len=10 cap=10) {
                      (uint32) 42719364,
                      (uint32) 16170776,
                      (uint32) 39276458,
                      (uint32) 15419045,
                      (uint32) 54940626,
                      (uint32) 39371554,
                      (uint32) 34231400,
                      (uint32) 62907004,
                      (uint32) 6482098,
                      (uint32) 2100676
                    }
                  },
                  y: (secp256k1.FieldVal) {
                    n: ([10]uint32) (len=10 cap=10) {
                      (uint32) 27378110,
                      (uint32) 9865392,
                      (uint32) 50819165,
                      (uint32) 8822040,
                      (uint32) 34618901,
                      (uint32) 4932053,
                      (uint32) 21273158,
                      (uint32) 41733215,
                      (uint32) 40201822,
                      (uint32) 1232818
                    }
                  }
                }),
                TweakedScriptKey: (*asset.TweakedScriptKey)(0xc000936cc0)({
                  RawKey: (keychain.KeyDescriptor) {
                    KeyLocator: (keychain.KeyLocator) {
                      Family: (keychain.KeyFamily) 212,
                      Index: (uint32) 8
                    },
                    PubKey: (*secp256k1.PublicKey)(0xc0008aad70)({
                      x: (secp256k1.FieldVal) {
                        n: ([10]uint32) (len=10 cap=10) {
                          (uint32) 7053852,
                          (uint32) 25782209,
                          (uint32) 54931100,
                          (uint32) 24052893,
                          (uint32) 13335950,
                          (uint32) 60446638,
                          (uint32) 11214322,
                          (uint32) 36176660,
                          (uint32) 20154626,
                          (uint32) 2377012
                        }
                      },
                      y: (secp256k1.FieldVal) {
                        n: ([10]uint32) (len=10 cap=10) {
                          (uint32) 61829780,
                          (uint32) 36292997,
                          (uint32) 13545354,
                          (uint32) 25077049,
                          (uint32) 22803081,
                          (uint32) 370001,
                          (uint32) 22785380,
                          (uint32) 9586648,
                          (uint32) 24018446,
                          (uint32) 3111824
                        }
                      }
                    })
                  },
                  Tweak: ([]uint8) <nil>,
                  Type: (asset.ScriptKeyType) 1
                })
              },
              GroupInternalKey: (*keychain.KeyDescriptor)(<nil>),
              GroupTapscriptRoot: ([]uint8) <nil>,
              ExternalKey: (fn.Option[github.com/lightninglabs/taproot-assets/asset.ExternalKey]) {
                isSome: (bool) false,
                some: (asset.ExternalKey) {
                  XPub: (hdkeychain.ExtendedKey) {
                    key: ([]uint8) <nil>,
                    pubKey: ([]uint8) <nil>,
                    chainCode: ([]uint8) <nil>,
                    depth: (uint8) 0,
                    parentFP: ([]uint8) <nil>,
                    childNum: (uint32) 0,
                    version: ([]uint8) <nil>,
                    isPrivate: (bool) false
                  },
                  MasterFingerprint: (uint32) 0,
                  DerivationPath: ([]uint32) <nil>
                }
              }
            })
        planter_test.go:387: Batch seedling mismatch:
            actual seedlings: (map[string]*tapgarden.Seedling) (len=3) {
             (string) (len=64) "e1d858948792df736600505393d11717213a213d8c10d13c96ecbe79e6948b57": (*tapgarden.Seedling)(0xc0007d0420)(AssetSeedling(name=e1d858948792df736600505393d11717213a213d8c10d13c96ecbe79e6948b57, type=Normal, amt=1126390197, version=1) received),
             (string) (len=64) "81a0ade12d13b0a4b98f7a0d265743863e4604980e7486447ec566f8844fdd58": (*tapgarden.Seedling)(0xc0007d0160)(AssetSeedling(name=81a0ade12d13b0a4b98f7a0d265743863e4604980e7486447ec566f8844fdd58, type=Normal, amt=1252564738, version=1) received),
             (string) (len=64) "3fb5ec0072d0a219880c550e4b7d15a597266b2b2c61c04c62eee197599be754": (*tapgarden.Seedling)(0xc0007d02c0)(AssetSeedling(name=3fb5ec0072d0a219880c550e4b7d15a597266b2b2c61c04c62eee197599be754, type=Normal, amt=182085842, version=0) received)
            }

            expected seedlings: ([]*tapgarden.Seedling) (len=4 cap=4) {
             (*tapgarden.Seedling)(0xc0008c6840)(AssetSeedling(name=65fff78845e603d0788ee37a622e6f28248c433fbd70691dafb6dcb8754c7b76, type=Collectible, amt=1, version=0) received),
             (*tapgarden.Seedling)(0xc0008c69a0)(AssetSeedling(name=81a0ade12d13b0a4b98f7a0d265743863e4604980e7486447ec566f8844fdd58, type=Normal, amt=1252564738, version=1) received),
             (*tapgarden.Seedling)(0xc0008c6b00)(AssetSeedling(name=3fb5ec0072d0a219880c550e4b7d15a597266b2b2c61c04c62eee197599be754, type=Normal, amt=182085842, version=0) received),
             (*tapgarden.Seedling)(0xc0008c6c60)(AssetSeedling(name=e1d858948792df736600505393d11717213a213d8c10d13c96ecbe79e6948b57, type=Normal, amt=1126390197, version=1) received)
            }

            error: expected 4 seedlings, got 3
```

The full log from that run can be seen [here](https://github.com/user-attachments/files/23724364/issue1881_flake.log).

--- a/tapgarden/planter_test.go
+++ b/tapgarden/planter_test.go
@@ -591,6 +591,11 @@ func (t *mintingTestHarness) finalizeBatchAssertFrozen(
 	t.finalizeBatch(&wg, respChan, nil)
 
 	if noBatch {
+		// Wait for the FinalizeBatch goroutine to complete before
+		// returning. This prevents a race where the goroutine's
+		// stateReqs message could interleave with subsequent seedling
+		// requests, potentially corrupting batch state.
+		wg.Wait()
 		t.assertNoPendingBatch()
 		return nil
 	}

